c++/src/kj/async-io-unix.c++

```diff
@@ -478,6 +485,19 @@ public:
       return array.finish();
     }
 
+    if (str.startsWith("unix-abstract:")) {
+      StringPtr path = str.slice(strlen("unix-abstract:"));
+      KJ_REQUIRE(path.size() + 1 < sizeof(addr.unixDomain.sun_path),
+                 "Unix domain socket address is too long.", str);
+      result.addr.unixDomain.sun_family = AF_UNIX;
+      result.addr.unixDomain.sun_path[0] = '\0';
+      memcpy(result.addr.unixDomain.sun_path + 1, path.cStr(), path.size());
```

Comment:
Since you use memcpy() here rather than strcpy(), the NUL terminator doesn't get copied. I'm not sure if Linux requires the string to be terminated (since it knows the addrlen separately), but it `toString()` you appear to be expecting a NUL-terminated string. Probably to be save we should assume a NUL terminator is required?

It would be nice to have a test that actually binds to or connects to an abstract address, ideally in a way that proves resolution is not filesystem-dependent... could be accomplished with some chdir()s. If you aren't interested in writing this test, though, don't worry about it.